src/util/io.c

```diff
@@ -140,33 +143,55 @@ socket_close (
     return TSS2_RC_SUCCESS;
 }
 
+static void uinttostring(unsigned int num, char *str) {
+    int len = 0;
+    unsigned int n;
+ 
+    for (n = num; n != 0; ++len, n /= 10) ;
+
+    for (int i = 1, n = num; i <= len; n /= 10, ++i) {
+        str[len - i] = '0' + (n % 10);
+    }
+    str[len] = '\0';
+}
+
 TSS2_RC
 socket_connect (
     const char *hostname,
     uint16_t port,
     SOCKET *sock)
 {
-    struct sockaddr_in sockaddr = { 0 };
+    static const struct addrinfo hints = { .ai_socktype = SOCK_STREAM,
+        .ai_protocol = IPPROTO_TCP };
+    struct addrinfo *retp = NULL;
+    struct addrinfo *p;
+    char port_str[MAX_PORT_STR_LEN];
     int ret = 0;
 
-    LOG_DEBUG ("Creating AF_INET stream socket");
-    *sock = socket (AF_INET, SOCK_STREAM, IPPROTO_TCP);
-    if (*sock == -1) {
-        LOG_WARNING ("Failed to create socket. errno %d: %s",
-                     errno, strerror (errno));
+    uinttostring(port, port_str);
+
+    LOG_DEBUG ("Resolving host %s", hostname);
+    ret = getaddrinfo (hostname, port_str, &hints, &retp);
+    if (ret != 0) {
+        LOG_WARNING ("Host %s does not resolve to a valid address: %d: %s",
+            hostname, ret, gai_strerror(ret));
         return TSS2_TCTI_RC_IO_ERROR;
     }
-    sockaddr.sin_family = AF_INET;
-    sockaddr.sin_addr.s_addr = inet_addr (hostname);
-    sockaddr.sin_port = htons (port);
-
-    LOG_DEBUG ("Connecting socket %d to hostname %s on port %" PRIu16,
-               *sock, hostname, port);
-    ret = connect (*sock, (struct sockaddr*)&sockaddr, sizeof (sockaddr));
-    if (ret == -1) {
+    
+    LOG_DEBUG ("Connecting TCP stream socket to host %s on port %" PRIu16,
```

Comment:
Please put some the LOG_DEBUG with the actual target node inside the loop instead. I think that gives way more information...

If you set the hints.ai_flags = AI_CANONIDN then p->ai_canonname should also be set correctly. Never used it myself, but should work...

Reply:
I added it, but, although it is useful, be warned that AI_CANONIDN is a non-standard glib-specific extension.

Reply:
OK. Because AI_CANONIDN is non-standard. It is a glib-specific extension, I had to add a #define __USE_GNU to break the standards enforcement in netdb.h.

Reply:
Well the actual point was to bring this LOG-Statement inside the `for`-loop, so we see which IPs the lib is attempting to connect to...
I guess a __USE_GNU addition would not be the nicest...
Google gives us an alternative: `inet_ntop` see below...